dpnp/dparray.pyx

```diff
@@ -497,26 +497,15 @@ cdef class dparray:
         if not utils.use_origin_backend(self):
             c_order, fortran_order = self.flags.c_contiguous, self.flags.f_contiguous
 
-            if order not in {'C', 'F', 'A', 'K'}:
+            if not isinstance(self, dparray):
+                pass
```

Comment:
Do you really need this extra check?
`self` represents the instance of the class `dparray`, and `flatten` is a method of that class.
It is impossible that `isinstance(self, dparray)` is false.

Reply:
Remove this extra check please.